scintillometry/io/psrfits/hdu.py

```diff
@@ -67,17 +67,43 @@ def location(self):
         except KeyError:
             return None
 
+    @location.setter
+    def location(self, loc):
+        """ Location setter. input should be an Astropy EarthLocation"""
+        # TODO, for the space base observatory, Earth Location may not apply.
+        self.hdu.header['ANT_X'] = loc.itrs.x.to_value(u.m)
+        self.hdu.header['ANT_Y'] = loc.itrs.y.to_value(u.m)
+        self.hdu.header['ANT_Z'] = loc.itrs.z.to_value(u.m)
+
     @property
     def start_time(self):
         return (Time(self.header['STT_IMJD'], format='mjd', precision=9,
                      location=self.location) +
                 TimeDelta(self.header['STT_SMJD'], self.header['STT_OFFS'],
                           format='sec', scale='tai'))
 
+    @start_time.setter
+    def start_time(self, time):
+        """ Set the start_time, the input value should be an Time object"""
+        if not time.isscalar:
```

Comment:
I'd remove this check, the next line will fail with essentially the same error message anyway.